### PR TITLE
added compatibility for esxi 3rd party distributed switches

### LINKED ISSUE #2058: ESXI 3rd party distributed switches not finding VM IP
There are cases where 3rd party distributed switches do not respond correctly to the `esxcli network vm port list` and can only be queried by issuing a `vim-cmd vmsvc/get.guest` command.

### PR DESCRIPTION
**Pull request by tracyde**
Fixes #2058

### CODE REVIEW

File: builder/vmware/iso/driver_esx5.go [hunk 1]
```diff
@@ -252,7 +246,43 @@ func (d *ESX5Driver) SSHAddress(state multistep.StateBag) (string, error) {
 		return "", errors.New("VM network port found, but no IP address")
 	}
 
-	address := fmt.Sprintf("%s:%d", record["IPAddress"], config.SSHPort)
+	return record["Address"], nil
```
Comment (markpeek) [inline builder/vmware/iso/driver_esx5.go]: Shouldn't this be ```return record["IPAddress"], nil```? (Address => IPAddress)

Comment (tracyde) [inline builder/vmware/iso/driver_esx5.go]: You are right.

**Comment by markpeek** [comment]
@tracyde your PR needed to be merged up with master which I did [here](https://github.com/markpeek/packer/tree/tracyde-iss2058). Can you verify these are the right changes?

**Comment by tracyde** [comment]
@markpeek Sorry, I didn't get a notification of the comment. Those are the correct changes.

**Comment by makeacode** [comment]
@tracyde Can we revisit merging this PR?  This is still preventing us from upgrading beyond 0.8.6 when we patched it last

**Comment by mwhooker** [comment]
see #4444

**Pull request closed without merging**